### PR TITLE
feat:refactor story readOne

### PR DESCRIPTION
**Pull request by izquierdoalfredo**
https://github.com/SeedCompany/cord-api-v3/issues/1019

### CODE REVIEW

File: src/components/story/story.service.ts [hunk 1]
```diff
@@ -204,71 +208,67 @@ export class StoryService {
...
       scriptureReferences: {
-        canRead: !!result.story.canScriptureReferencesRead,
-        canEdit: !!result.story.canScriptureReferencesEdit,
+        canRead: (perms as any)?.scriptureReferences?.canRead,
+        canEdit: (perms as any)?.scriptureReferences?.canEdit,
         value: scriptureReferences,
       },
-      createdAt: result.story.createdAt,
+      id: result.node.properties.id,
+      createdAt: result.node.properties.createdAt,
```
[the remark this answers is not in the document]
Comment (izquierdoalfredo) [reply src/components/story/story.service.ts]: Yes, you are right. I removed these lines.

File: src/components/story/story.service.ts [hunk 2]
```diff
@@ -204,71 +208,67 @@ export class StoryService {
...
-        mapping('story', {
-          ...addBaseNodeMetaPropsWithClauseAsObject(baseNodeMetaProps),
-          ...listWithSecureObjectAsObject(secureProps),
-          canReadStorys: 'requestingUser.canReadStorys',
-          canScriptureReferencesRead: 'scriptureReferencesReadPerm.read',
-          canScriptureReferencesEdit: 'scriptureReferencesEditPerm.edit',
-        })
-      );
+      .with('{value: props.value, property: type(r)} as prop, permList, node')
+      .with('collect(prop) as propList, permList, node')
```
[the remark this answers is not in the document]
Comment (izquierdoalfredo) [reply src/components/story/story.service.ts]: I refactored it

File: src/components/story/story.service.ts [hunk 3]
```diff
@@ -204,71 +208,67 @@ export class StoryService {
...
     return {
-      id: result.story.id,
-      name: result.story.name,
+      ...parseBaseNodeProperties(result.node),
+      ...securedProps,
       scriptureReferences: {
-        canRead: !!result.story.canScriptureReferencesRead,
-        canEdit: !!result.story.canScriptureReferencesEdit,
+        canRead: (perms as any)?.scriptureReferences?.canRead,
+        canEdit: (perms as any)?.scriptureReferences?.canEdit,
```
[the remark this answers is not in the document]
Comment (izquierdoalfredo) [reply src/components/story/story.service.ts]: "scriptureReferences" is not present in Story type. 
Can you provide me correct way to use permissions here?

[the remark this answers is not in the document]
Comment (CarsonF) [reply src/components/story/story.service.ts]: @izquierdoalfredo https://github.com/SeedCompany/cord-api-v3/pull/1034 should fix your problem here. After rebasing on master, you should be able to use `user: true` in `parseSecureProperties` and then spread it into the result.

**Comment by izquierdoalfredo** [comment]
Thanks for your help @CarsonF

**Pull request merged**